search-app/src/cmr/search/services/association_validation.clj

```diff
@@ -170,15 +170,8 @@
                    "please delete one of the conflicting generic associations.")
               concept-id (:concept-id association)))))
 
-(defmulti validate-association-conflict-for-collection
-  "Validate the given association does not conflict with existing tag associations in that
-  a tag/variable cannot be associated with a collection revision and the same collection
-  without revision at the same time."
-  (fn [context assoc-type assoc-id association]
-    assoc-type))
-
```

Comment:
Why do you want to remove the defmulti and define individual ones? I don't like defmulti either but isn't it what it's supposed to be used for?

Reply:
Case is best if the choices can be determined by the compiler then cond, then condp.  We should only be using defmulti if we truly need polymorphism.  Using defmulti takes much longer when the software is running (more processing time) to determine which function to call, so it is a lot more efficient to not use it. 

In the CMR we really don't need to use polymorphism, we are not writing a library that will be used by industry to computer other stuff where they will need to change the behavior of functions.